### PR TITLE
ensure docker running before running commands

### PR DESCRIPTION
**Pull request by phiSgr**
The failure message is this:

`Failed to run docker commands. Is docker running?`

### CODE REVIEW

**Comment by tg339** [comment]
can we create a cleaner error message with show_message

`Failed to run docker commands. Is docker running?`

File: apps/igloo-kit-cli/src/cli/routines.rs [hunk 1]
```diff
@@ -126,10 +127,10 @@ impl RoutineSuccess {
 pub struct RoutineFailure {
     message: Message,
     message_type: MessageType,
-    error: Error,
+    error: Option<Error>,
 }
 impl RoutineFailure {
-    pub fn new(message: Message, error: Error) -> Self {
+    pub fn new(message: Message, error: Option<Error>) -> Self {
```
Comment (phiSgr) [inline apps/igloo-kit-cli/src/cli/routines.rs]: This caused a big diff. Maybe I should create another function to create a `RoutineFailure` without an `Error`.

Comment (tg339) [reply apps/igloo-kit-cli/src/cli/routines.rs]: What we're essentially saying here is that there are cases where a routine fails to generate the expected result but didn't error out. Right?

Maybe we should rethink the return type of the routines? What comes to mind is  a result with three cases Success, Failure, Error

Comment (tg339) [reply apps/igloo-kit-cli/src/cli/routines.rs]: That wouldn't solve the diff issue thought 😛

Comment (tg339) [reply apps/igloo-kit-cli/src/cli/routines.rs]: I'm a bit curious as to what the use case for a failure that isn't an error. Maybe we need to rethink that?

**Review by tg339 — APPROVED** [review]

File: apps/igloo-kit-cli/src/cli/routines.rs (hunk 1 above)
Comment (phiSgr) [reply apps/igloo-kit-cli/src/cli/routines.rs]: > a failure that isn't an error

to be precise, a failure that isn't **caused by** an **IO** error

**Pull request merged**